contracts/drafts/issuance/VentureEth.sol

```diff
@@ -0,0 +1,70 @@
+pragma solidity ^0.5.10;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20Mintable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "../../issuance/IssuanceEth.sol";
+import "../../token/ERC20DividendableEth.sol";
+
+
+/**
+ * @title VentureEth
+ * @notice Implements a venture
+ *
+ * 1. Use `setIssuePrice` to determine how many ether (in wei) do investors
+ *    have to pay for each issued token.
+ * 2. Use `openIssuance` to allow investors to invest.
+ * 3. Investors can `invest` their ether at will.
+ * 4. Investors can also `cancelInvestment` and get their ether back.
+ * 5. The contract owner can `cancelAllInvestments` to close the investment phase.
+ *    In this case `invest` is not available, but `cancelInvestment` is.
+ * 6. Use `startDistribution` to close the investment phase.
+ * 7. Investors can only `withdraw` their issued tokens now.
+ * 8. Owner can use `transferFunds` to send collected ether to a wallet.
+ * 9.
+ * 10.
+ */
+contract VentureEth is ERC20Mintable, ERC20DividendableEth, IssuanceEth {
+
+    constructor() public ERC20Mintable()
+    ERC20DividendableEth() IssuanceEth(address(this)) {
+        _createState("ENDED");
+        _createTransition("LIVE", "ENDED");
+        _createTransition("FAILED", "ENDED");
+        addMinter(address(this));
+    }
+
+    modifier afterIssuance {
+        require(
+            currentState == "ENDED",
+            "Issuance must have ended."
+        );
+        _;
+    }
+
+    /**
+     * @notice Send ether to this function in orther to disburse dividends. Venture must be live.
+     */
+    function increasePool() external payable afterIssuance {
```

Comment:
I assume we are using `afterIssuance` here because otherwise the Ether collected during the issuance process gets mixed with the Ether injected in the contract for dividend distribution?

I would see logical that the Ether raised during the issuance remains in the VentureEth contract, to be used however the contract sees fit. Not sure if that would cause accounting problems, though.

Reply:
That was an accomodation for the `transferFunds` function. After an issuance, the owner of the issuance wants to use those funds. The problem was if he called transfer funds after the first client `increase`d`Pool`, than dividends could not be disbursed. It's mostly a guard for `transferFunds` to be called once, and `incresePool` or `updateAccount` only after `transferFunds` has been called.

Reply:
Take this scenario: Issuance raises 5 eth, the owner transfers all funds. Then, venture raises 5 eth in dividends, and no investor has called `updateAccount` yet, but the owner calls again `transferFunds` and all dividends are lost.

Reply:
I see. With this approach though investors can't `withdraw` once the Venture moves to `"ENDED"`.

What about just keeping track of the funds removed from the contract by the owner? Check this: https://github.com/HQ20/contracts/pull/162